File: CPP/Easy/Reverse-Integer.cpp

class Solution {
public:
int reverse(int x)
{
int sign = x<0?-1:1;
long rev=0;
int r;
int y = pow(2,31)-1;

    x = abs(x);
    
    while(x)
    {
        r = x%10;
        x = x/10;
        rev = rev*10 + r;
    }
    
    rev = rev*sign;
    
    if(rev<=y && rev>=(-1*y-1))
        return (int)rev;
    else 
        return 0;
}
};

File: Swift/Easy/Reverse-Integer.swift
class Solution {
    func reverse(_ x: Int) -> Int {
        
        var number = x
        var reverseNumber = 0
        
        while (number != 0) {
            let reminder = number % 10
            number /= 10
            reverseNumber *= 10
            reverseNumber += reminder
        }
        
        if reverseNumber > Int32.max || reverseNumber < Int32.min { return 0 }
        return reverseNumber
    }
}